engine/src/flutter/lib/web_ui/lib/src/engine/web_paragraph/paint.dart

```diff
@@ -95,6 +103,14 @@ class TextPaint {
       final int step = block.isLtr ? 1 : -1;
       for (int i = start; i != end; i += step) {
         final clusterText = layout.allClusters[i];
+        // If we have shadows, we need to make sure the canvas is big enough
+        final blockHasShadows = block.style.hasElement(StyleElements.shadows);
+        // We need to adjust the canvas size to fit the block in case there is scaling or zoom involved
+        final double zoomFactor = painter.adjustCanvas(
+          canvas,
+          clusterText.advance.width.ceilToDouble() + (blockHasShadows ? 200 : 0),
+          clusterText.advance.height.ceilToDouble() + (blockHasShadows ? 200 : 0),
```

Comment:
We shouldn't be sizing the canvas to each cluster size. Let's pick a high enough reasonable size that we know fits all clusters.